teps/0141-platform-context-variables.md

```diff
@@ -221,190 +232,319 @@ The downside of this approach is that Tasks can't easily reference the values of
+
+```json
+{
+    "predicateType": "https://slsa.dev/provenance/v1",
+    "predicate": {
+        "buildDefinition": {
+            "externalParameters": {
+                "runSpec": {
+                    "taskRef": {
+                        "name": "git-clone",
+                    },
+                    "params": [
+                        {
+                            "name": "repo_url",
+                            "value":  "$(context.platform.repo_url)"
+                        },
+                        {
+                            "name": "git-revision",
+                            "value":  "main"
+                        }
+                    ],
+                    // ...
+                }
+            },
+            "internalParameters": {
+              "tekton-pipelines-feature-flags": {
+                //...
+              },
+              "platform-parameters": {
+                "repo_url": "https://github.com/tektoncd/pipeline"
+              }
+            }
+        }
+    }
+}
+```
 
 ## Design Evaluation
-<!--
-How does this proposal affect the api conventions, reusability, simplicity, flexibility
-and conformance of Tekton, as described in [design principles](https://github.com/tektoncd/community/blob/master/design-principles.md)
--->
 
 ### Reusability
 
-<!--
-https://github.com/tektoncd/community/blob/main/design-principles.md#reusability
-
-- Are there existing features related to the proposed features? Were the existing features reused?
-- Is the problem being solved an authoring-time or runtime-concern? Is the proposed feature at the appropriate level
-authoring or runtime?
--->
+Remote Pipelines and Tasks should be able to use the platform variables, but if users want to make them reusable (e.g. publish on GitHub), the reference should be disallowed and the platform context variables need to be passed from `params` in PipelineRun or TaskRun. So the proposal doesn't break the reusability of current Tasks and Pipelines.
 
 ### Simplicity
 
-<!--
-https://github.com/tektoncd/community/blob/main/design-principles.md#simplicity
-
-- How does this proposal affect the user experience?
-- What’s the current user experience without the feature and how challenging is it?
-- What will be the user experience with the feature? How would it have changed?
-- Does this proposal contain the bare minimum change needed to solve for the use cases?
-- Are there any implicit behaviors in the proposal? Would users expect these implicit behaviors or would they be
-surprising? Are there security implications for these implicit behaviors?
--->
+This proposal requires less infrastructure for the platform builder to maintain than the alternative of a [webhook server](#webhook-responds-to-requests-for-context-parameters). It also results in a more consistent user experience for Task/Pipeline users, since all platforms that inject parameters will use the `$(context.platform.foo)` syntax rather than customized syntax.
```

Comment:
I'm not sure, I guess so. I basically copied most of the contents from Lee's original proposal. 😄 
I think if we choose to propose `$(context.platform.foo)`, it indeed adds the simplicity. (I can remove my customized syntax option since it conflicts with this :/)